src/index.js

```diff
@@ -0,0 +1,8 @@
+'use strict'
+
+module.exports = {
+  Stop: require('./circuit/hop'),
+  Dialer: require('./dialer'),
+  multicodec: require('./multicodec'),
+  tag: 'Circuit'
+}
```

Comment:
I also think that `transport-interface` is possibly a misnomer, perhaps we can rename it to something like connection handler/supplier, once you break away from the assumption that its a transport any possible impedance mismatches start going away.